lib/god/contacts/sensu.rb

```diff
@@ -0,0 +1,57 @@
+# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+# Send a notice to a SENSU client socket, port 3030 on 'localhost' only.
+# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+# [mandatory]
+# check_name    - a unique check name
+#
+# [optional]
+# status_code   - status codes used are 0 for OK, 1 for WARNING, 2 for CRITICAL, and 3 or greater to indicate UNKNOWN or CUSTOM.
+# handler       - default handler
+#
+
+CONTACT_DEPS[:sensu] = ['json']
+CONTACT_DEPS[:sensu].each do |d|
+  require d
+end
+
+module God
+  module Contacts
+
+    class Sensu < Contact
+      class << self
+        attr_accessor :check_name, :status_code, :handler, :host, :port
+      end
+
+      self.status_code = 2
+      self.handler = 'default'
+      self.host = 'localhost'
+      self.port = 3030
+
+      def valid?
+        valid = true
+        valid &= complain("Attribute 'check_name' must be specified", self) unless arg(:check_name)
+        valid
+      end
+
+      attr_accessor :check_name, :status_code, :handler, :host, :port
+
+      def sensu_client_socket(msg)
+        u = UDPSocket.new
+        u.send(msg + "\n", 0, arg(:host).nil? ? self.host : arg(:host), arg(:port).nil? ? self.port : arg(:port))
```

Comment:
You'll want to close the `UDPSocket` once you're done using it to prevent leaking sockets.